### PR TITLE
Add insights-client registration tests for rhel7 and rhel8

### PR DESCRIPTION
**Pull request by jameerpathan111**

### CODE REVIEW

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_rhcloud_insights.py

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_rhcloud_insights.py

**Comment by jameerpathan111** [comment]
See PRT job 219 for test result.

```
14:41:27  + py.test tests/foreman/cli/test_rhcloud_insights.py --junit-xml=sat-results.xml -o junit_suite_name=sat-result
14:41:32  ============================= test session starts ==============================
14:41:32  platform linux -- Python 3.8.6, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
14:41:32  shared_function enabled - OFF - scope:  - storage: file
14:41:32  rootdir: /opt/app-root/src/robottelo, configfile: pyproject.toml
14:41:32  plugins: forked-1.3.0, ibutsu-1.16, mock-3.6.1, reportportal-5.0.8, services-2.2.1, xdist-2.3.0
14:41:32  collected 2 items
14:41:32  
14:56:31  tests/foreman/cli/test_rhcloud_insights.py ..                            [100%]
14:56:31  ================== 2 passed, 3 warnings in 885.86s (0:14:45) ===================
```

File: tests/foreman/cli/test_rhcloud_insights.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        1. Run 'insights-client --test-connection'.
+
+    :expectedresults: 'insights-client --test-connection' should return 0.
+
+    :CaseImportance: Critical
+    """
+    org, activation_key = organization_ak_setup
+    with VMBroker(nick=distro, host_classes={'host': ContentHost}) as vm:
+        vm.configure_rhai_client(default_sat, activation_key.name, org.label, distro)
+        result = vm.run('insights-client --test-connection')
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_rhcloud_insights.py:49]: I would rather use the `vm.execute`

**Review by omkarkhatavkar — APPROVED** [review]
ACK, pending one suggestion

File: tests/foreman/cli/test_rhcloud_insights.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import pytest
+from broker.broker import VMBroker
+
+from robottelo.constants import DISTRO_RHEL7
+from robottelo.constants import DISTRO_RHEL8
+from robottelo.hosts import ContentHost
+
+
+@pytest.mark.tier4
+@pytest.mark.run_in_one_thread
```
Comment (devendra104) [inline tests/foreman/cli/test_rhcloud_insights.py]: I think it is not required.

File: tests/foreman/cli/test_rhcloud_insights.py (hunk 1 above)
Comment (jameerpathan111) [reply tests/foreman/cli/test_rhcloud_insights.py:48]: @omkarkhatavkar internally it's `run = Host.execute`. https://github.com/SatelliteQE/robottelo/blob/master/robottelo/hosts.py#L77

File: tests/foreman/cli/test_rhcloud_insights.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    """Verify that 'insights-client --test-connection' successfully tests the proxy connection via
+    the Satellite.
+
+    :id: 61a4a39e-b484-49f4-a6fd-46ffc7736e50
+
+    :Steps:
+
+        0. Create RHEL7 and RHEL8 VM and register to insights within org having manifest.
+
+        1. Run 'insights-client --test-connection'.
```
Comment (devendra104) [inline tests/foreman/cli/test_rhcloud_insights.py]: Steps would be good if it starts from 1 instead of 0.

Comment (jameerpathan111) [reply tests/foreman/cli/test_rhcloud_insights.py]: updated the pr.

**Review by devendra104 — APPROVED** [review]

**Comment by omkarkhatavkar** [comment]
waiting for the CI result

**Pull request merged**